draft-ietf-tls-tls13.md

```diff
@@ -4375,6 +4375,89 @@ currently deployed and properly configured TLS implementations.
 %%### Ticket Establishment
 %%### Updating Keys
 
+# State Machine
+
+This section provides a summary of the legal state transitions for the
+client and server handshakes.  State names (in all capitals, e.g.,
+START) have no formal meaning but are provided for ease of
+comprehension.  Messages which are sent only sometimes are indicated
+in [].
+
+## Client
+
+~~~~
+          +-----> START
+     Recv |        |
+    Hello |        | Send ClientHello
+    Retry |        v
+  Request +---- WAIT_SH            
+                   | Recv ServerHello
+                   V                 
+                WAIT_EE            
+                   | Recv EncryptedExtensions         
+          +--------+--------+          
+Using PSK |                 | Using Certificate             
+          |                 v
+          |            WAIT_CERT_CR
+          |        Recv |       | Recv CertificateRequest
+          | Certificate |       v
+          |             |    WAIT_CERT
+          |             |       | Recv Certificate
+          |             v       v
+          |              WAIT_CV
+          |                 | Recv CertificateVerify
+          +> WAIT_FINISHED <+
+                   | Recv Finished
+                   |
+                   | [Send EndOfEarlyData]
+                   | [Send Certificate [+ CertificateVerify]]
+                   | Send Finished
+                   v
+               CONNECTED
+~~~~
+
+## Server
+
+~~~~
+          +----> START  
+     Send |        |
+    Hello |        | Recv ClientHello
+    Retry |        |
+  Request |        v
+          +--- RECVD_CH
+                   | Select parameters
+                   v
+                NEGOTIATED
+                   | Send ServerHello
+                   | Send EncryptedExtensions
+                   | [Send CertificateRequest]
+                   | [Send Certificate + CertificateVerify]
+                   | Send Finished
+          +--------+--------+
+ No 0-RTT |                 | 0-RTT
+          |                 v
+          |             WAIT_EOED <---+
+          |        Recv |       |     | Recv
+          |        EOED |       |     | Application Data
```

Comment:
It's weird that this message (EndOfEarlyData) is written unabbreviated then abbreviated. I know that it takes a lot of room though :)